### PR TITLE
Proper support for the `duration` format

### PR DESCRIPTION
**Pull request by lukaszachy**
We claim to support same format as `sleep` commands so let's actually do it.

* [x] implement the feature
* [x] extend the test coverage
* [x] update specification
* [x] modify json schema

### CODE REVIEW

**Review by happz — APPROVED** [review]

**Review by psss — CHANGES_REQUESTED** [review]
Looks good, we just need to extend the schema as well:

    warn: /tests/core/smoke:duration - '1s 2s' does not match '^[0-9]+[smhd]$'

Proposing also a minor docs enhancement in 4a3ef7554962b2e7c7264c8ca3ba1f63092ee2e8.

File: tmt/utils.py [hunk 1]
```diff
@@ -2091,14 +2091,13 @@ def duration_to_seconds(duration: str) -> int:
...
-        match = re.match(r'^(\d+)([smhd]?)$', str(duration))
-        if match is None:
-            raise SpecificationError(f"Invalid duration '{duration}'.")
-        number, suffix = match.groups()
-        return int(number) * units.get(suffix, 1)
-    except (ValueError, AttributeError):
+    if re.match(r'^(\d+ *?[smhd]? *)+$', str(duration)) is None:
         raise SpecificationError(f"Invalid duration '{duration}'.")
+    total_time = 0
+    for part in re.findall(r'(\d+) *([smhd]?)', str(duration)):
```
Comment (KwisatzHaderach) [inline tmt/utils.py:2097]:
```suggestion
    for number, suffix in re.findall(r'(\d+) *([smhd]?)', str(duration)):
```
then you can skip the following line

File: tests/unit/test_utils.py [hunk 2]
```diff
@@ -225,8 +225,18 @@ def test_duration_to_seconds():
...
     assert duration_to_seconds('5h') == 18000
     assert duration_to_seconds('5d') == 432000
+    assert duration_to_seconds('5d') == 432000
+    # `man sleep` says: Given two or more arguments, pause for the amount of time
+    # specified by the sum of their values.
+    assert duration_to_seconds('1s 2s') == 3
+    assert duration_to_seconds('1h 2 3m') == 3600 + 2 + 180
+    # Divergence from 'sleep' as that expects space separated arguments
+    assert duration_to_seconds('1s2s') == 3
+    assert duration_to_seconds('1 m2   m') == 180
```
[the remark this answers is not in the document]
Comment (lukaszachy) [reply tests/unit/test_utils.py:235]: Exactly. The outcome of adjust / inheritance doesn't need to be human readable (messing with spaces is pain) And when we loose the human readability requirement.... IF someone wants to write it this way, well ... maybe we could add hook into tmt test show the final human readable value.

File: tmt/utils.py (hunk 1 above)
Comment (lukaszachy) [reply tmt/utils.py]: Right. done in 6dc708db

**Review by psss — APPROVED** [review]
Thanks for the nice fix & improvement! Spec example simplified in 16dc695a.

**Review by KwisatzHaderach — APPROVED** [review]

**Pull request merged**